enterprise/views.py

```diff
@@ -160,6 +165,34 @@ def render_page_with_error_code_message(request, context_data, error_code, log_m
     )
 
 
+def get_create_ent_enrollment(
+        course_id,
+        enterprise_customer_user,
+):
+    """
+    Get or Create the Enterprise Course Enrollment.
+
+    This method will set the Source if the Enterprise Course Enrollment (ECE)
+    was already created. It is possible the ECE will have been created by the
+    background task at the time of enrollment so we need to update the source
+    in that case.
+    """
+    source = EnterpriseEnrollmentSource.get_source(EnterpriseEnrollmentSource.ENROLLMENT_URL)
```

Comment:
I assume we know that the source should always be _ENROLLMENT_URL_, even for enrollments that already exist and that we're updating?

Reply:
When the views that use this method are called we want it to be `ENROLLMENT_URL` if it has not already been set.

There are cases where we create the CourseEnrollment before getting here that do not know what the Source is, so we update the Source and set it in this code if it is Null.